lib/IRGen/AllocStackHoisting.cpp

```diff
@@ -349,11 +363,9 @@ void HoistAllocStack::collectHoistableInstructions() {
         continue;
       }
       // Don't perform alloc_stack hoisting in functions with availability.
-      if (auto *Apply = dyn_cast<ApplyInst>(&Inst)) {
-        if (Apply->hasSemantics("availability.osversion")) {
-          AllocStackToHoist.clear();
-          return;
-        }
+      if (indicatesAvailabilityMacroUse(&Inst)) {
```

Comment:
Nitpick: I don't think you want to use the word "Macro" here. Availability macros are already something in Apple SDKs, and they aren't the dynamic checks that `#available` or ObjC `@available` do.